pkg/agent/packetcapture/packetcapture_controller_linux.go

```diff
@@ -0,0 +1,852 @@
+			if err != nil {
+				return fmt.Errorf("get PacketCapture failed: %w", err)
+			}
+			klog.InfoS("PacketCapture timeout", "name", pc.Name)
+			err = errors.New(captureTimeoutReason)
+			if updateErr := c.updatePacketCaptureStatus(pc.Name, captureState.capturedPacketsNum, "", err); updateErr != nil {
+				klog.ErrorS(updateErr, "The PacketCapture timed out, but PacketCapture status update failed", "name", pc.Name)
+			}
+			return err
+		}
+	}
+}
+
+func (c *Controller) getPodIP(podRef *crdv1alpha1.PodReference) (net.IP, error) {
+	podInterfaces := c.interfaceStore.GetContainerInterfacesByPod(podRef.Name, podRef.Namespace)
+	var result net.IP
+	if len(podInterfaces) > 0 {
+		result = podInterfaces[0].GetIPv4Addr()
+	} else {
+		pod, err := c.kubeClient.CoreV1().Pods(podRef.Namespace).Get(context.TODO(), podRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to get Pod %s/%s: %w", podRef.Namespace, podRef.Name, err)
+		}
+		podIPs := make([]net.IP, len(pod.Status.PodIPs))
+		for i, ip := range pod.Status.PodIPs {
+			podIPs[i] = net.ParseIP(ip.IP)
+		}
+		result = util.GetIPv4Addr(podIPs)
+	}
+	if result == nil {
+		return nil, fmt.Errorf("cannot find IP with IPV4 AddressFamily for Pod %s/%s", podRef.Namespace, podRef.Name)
+	}
+	return result, nil
+}
+
+func (c *Controller) parseIPs(pc *crdv1alpha1.PacketCapture) (srcIP, dstIP net.IP, err error) {
+	if pc.Spec.Source.Pod != nil {
+		srcIP, err = c.getPodIP(pc.Spec.Source.Pod)
+	} else if pc.Spec.Source.IP != nil {
+		srcIP = net.ParseIP(*pc.Spec.Source.IP)
+		if srcIP == nil {
+			err = errors.New("invalid source IP address: " + *pc.Spec.Source.IP)
+		}
+	}
+
+	if pc.Spec.Destination.Pod != nil {
+		dstIP, err = c.getPodIP(pc.Spec.Destination.Pod)
+	} else if pc.Spec.Destination.IP != nil {
+		dstIP = net.ParseIP(*pc.Spec.Destination.IP)
+		if dstIP == nil {
+			err = errors.New("invalid destination IP address: " + *pc.Spec.Destination.IP)
+		}
+	}
+	return
+}
+
+func (c *Controller) patchStatusIfNeeded(name string, status packetCapturePhase) error {
+	klog.InfoS("Patching PacketCapture CR Status if needed", "name", name, "status", status)
+	conditions := []crdv1alpha1.PacketCaptureCondition{}
+
+	if status == packetCapturePhaseRunning {
+		conditions = append(conditions, crdv1alpha1.PacketCaptureCondition{
+			Type:               crdv1alpha1.PacketCaptureRunning,
+			Status:             metav1.ConditionStatus(v1.ConditionTrue),
+			LastTransitionTime: metav1.Now(),
+		})
+	} else if status == packetCapturePhasePending || status == packetCapturePhaseUnknown {
+		conditions = append(conditions, crdv1alpha1.PacketCaptureCondition{
+			Type:               crdv1alpha1.PacketCapturePending,
+			Status:             metav1.ConditionStatus(v1.ConditionTrue),
+			LastTransitionTime: metav1.Now(),
+		})
+	} else {
+		return nil
+	}
+
+	toUpdate, getErr := c.packetCaptureLister.Get(name)
+	if getErr != nil {
+		return getErr
```

Comment:
Same question, I suppose we should ignore the not found error?   Otherwise, it will re-queue.

Reply:
`syncPacketCapture` will check not-found error.